src/CoreFoundation/CFReadStream.cs

```diff
@@ -140,15 +140,33 @@ protected override void UnscheduleFromRunLoop (CFRunLoop loop, NSString? mode)
 		}
 
 		[DllImport (Constants.CoreFoundationLibrary)]
+#if NET8_OR_GREATER
+		unsafe static extern byte CFReadStreamSetClient (/* CFReadStreamRef */ IntPtr stream, /* CFOptionFlags */ nint streamEvents,
+			/* CFReadStreamClientCallBack */ delegate* unmanaged<IntPtr, nint, IntPtr, void> clientCB, /* CFStreamClientContext* */ IntPtr clientContext);
+#else
 		[return: MarshalAs (UnmanagedType.I1)]
 		static extern bool CFReadStreamSetClient (/* CFReadStreamRef */ IntPtr stream, /* CFOptionFlags */ nint streamEvents,
 			/* CFReadStreamClientCallBack */ CFStreamCallback? clientCB, /* CFStreamClientContext* */ IntPtr clientContext);
+#endif
 
```

Comment:
Instead of `InvalidOperationException` which will fail at runtime, why not use `[Obsolete ("Use the other overload.")]` and catch this at compile time?

Reply:
I added an Obsolete attribute as well.